controller/rancher/rancher_events_handler.go

```diff
@@ -0,0 +1,243 @@
+package rancher
+
+import (
+	"fmt"
+	"github.com/Sirupsen/logrus"
+	revents "github.com/rancher/event-subscriber/events"
+	"github.com/rancher/go-rancher/v2"
+	"github.com/rancher/lb-controller/config"
+	"sync"
+	"time"
+)
+
+type EventsHandler interface {
+	Subscribe() error
+}
+
+type REventsHandler struct {
+	CattleURL       string
+	CattleAccessKey string
+	CattleSecretKey string
+	DoOnEvent       func(*config.Endpoint) bool
+	CheckOnEvent    func(*config.Endpoint) bool
+	DoOnTimeout     func(*config.Endpoint)
+	PollStatus      func(*config.Endpoint) bool
+	EventMap        map[string]*revents.Event
+	EventMu         *sync.RWMutex
+}
+
+func (ehandler *REventsHandler) Subscribe() error {
+
+	eventHandlers := map[string]revents.EventHandler{
+		"target.drain": ehandler.handle,
+		"ping":         ehandler.handle,
+	}
+
+	router, err := revents.NewEventRouter("", 0, ehandler.CattleURL, ehandler.CattleAccessKey, ehandler.CattleSecretKey, nil, eventHandlers, "", 250, revents.DefaultPingConfig)
+	if err != nil {
+		return err
+	}
+	err = router.StartWithoutCreate(nil)
+	return err
+}
+
+func (ehandler *REventsHandler) handle(event *revents.Event, cli *client.RancherClient) error {
+	log := logrus.WithFields(logrus.Fields{
+		"eventName":  event.Name,
+		"eventID":    event.ID,
+		"resourceID": event.ResourceID,
+	})
+	log.Infof("Rancher event: %#v", event)
```

Comment:
Does this log every ping event as well? If so, should be changed to not do that

Reply:
changed this to Debug level. Drain event will be logged at info by the handleDrainEvent anyway.